### DISCUSSION TITLE
Dynamic payload over MQTT

### DISCUSSION THREAD

**Discussion by unaiib** [apache/streampipes#3371] [Q&A]
We're exporting data with Apache Streampipes over MQTT, v0.95.1. We need to send only the signals that actually change their values (Report By Exception). Is it possible to implement this in the application?

We have tried to use the "Value Change" pipeline element, but doesn't work properly.

![image](https://github.com/user-attachments/assets/29360809-dca0-4391-a4d8-decbd62d0c13)

**Comment by tenthe** [comment]
Hi @unaiib,

It seems that we do not have a processing element for this specific use case. I think it would be useful to have such a processor. If you want, you can make a detailed description of the function in an issue and we can see if someone in the community implements the functionality. You are also welcome to write the extension yourself. You can find a tutorial here [1].

There are also generic processors such as the JavaScript Evaluator, in which you can implement your own scripts, but I think this might be a bit too difficult for your specific use case. 

Please feel free to contact me if you have any further questions.


Kind regards,
Philipp

[1] https://streampipes.apache.org/docs/extend-tutorial-data-processors/
